kolibri/content/serializers.py

```diff
@@ -261,6 +263,44 @@ class Meta:
         list_serializer_class = ContentNodeListSerializer
 
 
+class ContentNodeGranularSerializer(serializers.ModelSerializer):
+    total_resources = serializers.SerializerMethodField()
+    resources_on_device = serializers.SerializerMethodField()
+    importable = serializers.SerializerMethodField()
+
+    def get_total_resources(self, obj):
+        total_resources = obj.get_descendants(include_self=True).exclude(kind=content_kinds.TOPIC).count()
+
+        return total_resources
+
+    def get_resources_on_device(self, obj):
+        available_resources = obj.get_descendants(include_self=True).exclude(kind=content_kinds.TOPIC).filter(available=True).count()
+
+        return available_resources
+
+    def get_importable(self, obj):
+        if 'request' not in self.context or not self.context['request'].query_params.get('datafolder', None) or obj.kind == content_kinds.TOPIC:
+            return True
+        else:
+            datafolder = self.context['request'].query_params.get('datafolder', None)
+            files = FileSerializer(obj.files.all(), many=True, read_only=True).data
```

Comment:
Not sure why you need to instantiate a file serializer here, rather than querying the files directly?

`files = obj.files.all()`

Reply:
Yeah I think I should just query the files directly